x/crisis/handler.go

```diff
@@ -0,0 +1,80 @@
+package crisis
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// ModuleName is the module name for this module
+const (
+	ModuleName = "crisis"
+	RouterKey  = ModuleName
+)
+
+func NewHandler(k Keeper) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) sdk.Result {
+
+		switch msg := msg.(type) {
+		case MsgVerifyInvariant:
+			return handleMsgVerifyInvariant(ctx, msg, k)
+		default:
+			return sdk.ErrTxDecode("invalid message parse in crisis module").Result()
```

Comment:
`ErrTxDecode` is the wrong error "type" to return here.

Reply:
Can you elaborate? This code is consistent with all the modules

Reply:
In this case, somehow a message got routed incorrectly or the type is unsupported. It has nothing to do with tx decoding (at least not directly).

Reply:
hmm yes very good. opening an issue https://github.com/cosmos/cosmos-sdk/issues/3978